Character setup refactor.

PsiOmegaDelta opened the pull request:
Begins breaking down the large character setup in more manageable modules, allowing for easier additions, edits, and removals.
- [x] Ensure basic functionality.
- [x] Do **not** forget to switch from a shared global state, once client/preferences.dm has been cleaned up.
- [X] Port all settings into appropriate modules.
- [x] Figure out how to arrange each individual setting module, preferably automatically but in the worst case manually.

You can now load/save/reload your settings at any point, as long as the window is open, even if you have readied up.

![It's a start, sort of](https://www.dropbox.com/s/uaa3tlfm4wbxmjk/General.png?dl=1)

Things that should be possible once done:
- [ ] Changing your settings after joining the round.

GinjaNinja32 commented on code/modules/client/preference_setup/general/basic.dm:
```diff
@@ -0,0 +1,161 @@
+	S["skin_green"]			<< g_skin
+	S["skin_blue"]			<< b_skin
+	S["hair_style_name"]	<< h_style
+	S["facial_style_name"]	<< f_style
```
spaces for alignment please, otherwise it only works on 4-space tabs :(

GinjaNinja32 commented:
>Is there anything in HTML/CSS that resembles WrapPanel behavior from WPF/WinForms/etc.?

What's WrapPanel behaviour? Would a series of (possibly floating) divs work?

PsiOmegaDelta commented:
Basically what I would prefer is for blocks of HTML to flow from the top to the bottom, until they no longer fit.
When this happens, one repeats the process in a new column on the right (if necessary outside the right edge of the window), until all HTML-blocks have been iterated.

As for divs working, dunno.

atlantiscze commented:
Have you considered using Nano for the setup UI? I think it would look nicer, not sure how doable it is though.

PsiOmegaDelta commented:
I preferred to go with the safer option for the lobby screen.

PsiOmegaDelta commented:
However, I have to say NanoUI would be nicer in that I wouldn't have to recompile simply to test every little HTML/CSS change.

GinjaNinja32 commented:
I tried NanoUI lobby screens a while ago, you basically need a synchronous browse_rsc to know when the required things have been copied :(

PsiOmegaDelta committed:
Character setup refactoring.

Begins breaking down the large character setup in more manageable modules, allowing for easier additions, edits, and removals.

PsiOmegaDelta commented:
Minor progress update
[General Settings](https://www.dropbox.com/s/uaa3tlfm4wbxmjk/General.png?dl=1)
[Occupation Settings](https://www.dropbox.com/s/245exiobxr935d5/Occupation.png?dl=1)
[Antag Settings](https://www.dropbox.com/s/7zlinnkccwwi4lx/Antagonism.png?dl=1)
[Global Settings](https://www.dropbox.com/s/gihb4piesdo6ycn/Global.png?dl=1)

GinjaNinja32 commented:
:D

PsiOmegaDelta commented:
Finally got around to wrap up the important bits.
